### ISSUE TITLE
Language support: [Kotlin]

### ISSUE THREAD

**Issue by Mishyaz** [lppedd/idea-return-highlighter#14]
Kotlin plz

**Comment by ovitrif** [comment]
Please 👍

**Comment by yasandev** [comment]
Yeah, Kotlin support would be great. It's weird that the plugin doesn't support Kotlin even though looks like it was written in Kotlin it self.

**Comment by lppedd** [comment]
@ovitrif @yasandev I know it's strange but believe me when I say there is always a reason 😄
I'm doing my best to work and follow my personal projects but it's not easy. Hope you can understand

**Comment by ovitrif** [comment]
@lppedd I'll have a look myself, maybe I can help 😉, thanks already for your effort!

**Comment by yasandev** [comment]
@lppedd Thanks, yeah I understand, no pressure - I will also take a look to try adding the feature myself

**Comment by lppedd** [comment]
@ovitrif @yasandev if you have some spare time to try it out it would be cool!
Maybe I've abstracted things too much so you might need to fiddle around a bit before understanding how each piece work together with the others. If you think it requires too much time just let me know.
The issue is that return points in Kotlin are not as obvious as in other languages (in some cases there no `return` keyword for example), plus I'm back to Eclipse for the moment (working with [Rational Developer for i](https://www.ibm.com/products/rational-developer-for-i)) and the mental switch is a PITA

**Comment by yasandev** [comment]
Yeah, I actually took a look before commenting here and didn't understand how it worked - but probably will do after I look at the code for long enough.

I will try working on it but I won't really make any promises since I dont think I can put that much time into it, to be honest.

**Comment by demoth** [comment]
Linking other Kotlin requests: #28 and #29
